packages/js/product-editor/src/hooks/use-validation/README.md

```diff
@@ -0,0 +1,41 @@
+# useValidation
+
+This custom hook uses the helper functions `const { lockPostSaving, unlockPostSaving } = useDispatch( 'core/editor' );` to lock/unlock the current editing product before saving it.
+
+## Usage
+
+Syncronous validation
+
+```typescript
```

Comment:
I think at the DM we may have mentioned avoiding using TS for readmes and examples and documentation to reduce the barrier to entry for 3PD, but we can update this in a follow-up if that's how we want to handle this.  No need to hold up this PR.